app/services/analytics.rb

```diff
@@ -95,12 +96,36 @@ def browser_attributes
   end
 
   def session_duration
-    @session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
+    session[:session_started_at].present? ? Time.zone.now - session_started_at : nil
   end
 
   def session_started_at
-    value = @session[:session_started_at]
+    value = session[:session_started_at]
     return value unless value.is_a?(String)
     Time.zone.parse(value)
   end
+
+  def sp_request_attributes
+    resolved_result = resolved_authn_context_result
+    return resolved_result if resolved_result.nil?
+
+    attributes = resolved_result.to_h
+    attributes[:component_values] = resolved_result.component_values.map do |v|
+      v.to_h.slice(:name, :description)
```

Comment:
IMO the description seems redundant and more likely to keep inflating the size of the log payload?

Reply:
Sounds good - done in [b215e6e](https://github.com/18F/identity-idp/pull/10265/commits/b215e6e8dd3dcd1cc7379f1b194a5382e892ae79)